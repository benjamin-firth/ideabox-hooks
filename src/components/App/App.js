import React, { useState, useEffect } from 'react';
import './App.css';
import CardContainer from '../CardContainer/CardContainer';
import Form from '../Form/Form';

const App = () => {

  const [reservations, setReservations] = useState([]);

  const deleteReservation = (id) => {
    return fetch(`http://localhost:3001/api/v1/reservations/${id}`, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
      }
    })
    .then(response => response.json())
    .then(data => setReservations(data))
  }

  const addNewReservation = (name, date, time, number) => {
    fetch('http://localhost:3001/api/v1/reservations', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        name: name,
        date: date,
        time: time,
        number: number
      })
    })
    .then(response => response.json())
    .then(data => {
      return setReservations([...reservations, data]);
    })
  }

  useEffect(() => {
    fetch('http://localhost:3001/api/v1/reservations')
      .then(res => res.json())
      .then(data => setReservations(data));
  }, []);

  return (
    <div className="App">
      <h1 className='app-title'>Turing Cafe Reservations</h1>
      <div className='resy-form'>
      </div>
      <div className='enter-form'>
        <Form addNewReservation={addNewReservation} />
      </div>
      <div className='resy-container'>
        <CardContainer deleteReservation={deleteReservation} ogReservations={reservations} />
      </div>
    </div>
  )
}

export default App;
