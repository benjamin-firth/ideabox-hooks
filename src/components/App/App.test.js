import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  describe('App Snapshots', () => {
    it('Should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })
  })

})