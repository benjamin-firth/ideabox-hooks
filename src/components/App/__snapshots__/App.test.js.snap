// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Snapshots Should match the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  />
  <div
    className="enter-form"
  >
    <Form
      addNewReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <CardContainer
      ogReservations={Array []}
    />
  </div>
</div>
`;
